src/js/components/modals/ServiceFormModal.js

```diff
@@ -0,0 +1,244 @@
+      this.setState({
+        service: ServiceUtil.createServiceFromFormModel(model),
+        model: model,
+        errorMessage: null
+      });
+    }
+  }
+
+  handleClearError() {
+    this.setState({
+      errorMessage: null
+    });
+  }
+
+  handleJSONChange(app) {
+    let {service} = this.state;
+    try {
+      service = new Service(JSON.parse(app));
+    } catch (e) {
+
+    }
+    this.setState({app, service})
+  }
+
+  handleJSONToggle() {
+    if (!this.state.json) {
+      let model = this.triggerAdvancedSubmit().model;
+      let service = ServiceUtil.createServiceFromFormModel(model)
+      this.setState({
+        app: JSON.stringify(ServiceUtil
+          .getAppDefinitionFromService(service), null, 2),
+        model,
+        service
+      });
+    }
+    this.setState({json: !this.state.json});
+  }
+
+  onMarathonStoreServiceCreateSuccess() {
+    this.resetState();
+    this.props.onClose();
+  }
+
+  onMarathonStoreServiceCreateError(errorMessage) {
+    this.setState({
+      errorMessage
+    });
+  }
+
+  handleCancel() {
+    this.props.onClose();
+  }
+
+  handleSubmit() {
+    if (this.state.json) {
+      MarathonStore.createService(JSON.parse(this.state.app));
+      return ;
+    }
+    MarathonStore.createService(ServiceUtil.getAppDefinitionFromService(this.state.service));
+  }
+
+  getAdvancedSubmit(triggerSubmit) {
+    this.triggerAdvancedSubmit = triggerSubmit;
+  }
+
+  getErrorMessage() {
+    let {errorMessage} = this.state;
+    if (!errorMessage) {
+      return null;
+    }
+    return (
+      <div>
+        <h4 className="text-align-center text-danger flush-top">
+          {errorMessage.message}
+        </h4>
+        <pre className="text-danger">
+          {JSON.stringify(errorMessage.details, null, 2)}
+        </pre>
+        <button onClick={this.handleClearError}
+          className="button button-small button-danger flush-bottom">
+          clear
+        </button>
+      </div>
+    );
+  }
+
+  getFooter() {
+    return (
+      <div className="button-collection flush-bottom">
+        <button
+          className="button button-large flush-top flush-bottom"
+          onClick={this.handleCancel}>
+          Cancel
+        </button>
+        <ToggleButton
+          checked={this.state.json}
+          onChange={this.handleJSONToggle}>
+          JSON mode
+        </ToggleButton>
+        <button
+          className="button button-large button-success flush-bottom"
+          onClick={this.handleSubmit}>
+          Deploy
+        </button>
+      </div>
+    );
+
+  }
+
+  getModalContents() {
+    if (this.state.json) {
+      return (
+        <Ace editorProps={{$blockScrolling: true}}
+          mode="json"
+          onChange={this.handleJSONChange}
+          showGutter={true}
+          showPrintMargin={false}
+          theme="monokai"
+          value={this.state.app}
+          width="100%"/>
+      );
+    }
+
+    return (
+      <AdvancedConfig
```

Comment:
I think we can use `SchemaForm` here directly.
Actually looking at `AdvancedConfig` now, it is only a wrapper for `SchemaForm`. I think we should try to remove it all together, but not in this PR.